### PR TITLE
WT-8043 Tidy the "ripcip" argument into the visibility code

### PR DESCRIPTION
**Pull request by sauclovian-wt**
The "ripcip" argument into the visibility mechanism can just be "rip".

It is only used by the row code. There was one case where it was being
used as "cip", but that code was wrong, and (fortunately, then) also
unreachable. ("ins" is never null for column-store.)

Change it to be a WT_ROW * instead of a void *. It turns out that the
corresponding entry in WT_SAVE_UPD was already a WT_ROW *, so just
rename that (ripcip -> rip) for clarity.

It turns out this also allows removing the unsafe casts in WT_ROW_SLOT
and WT_COL_SLOT.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -1421,7 +1421,8 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
             continue;
 
         if (supd->ins == NULL) {
-            slot = WT_ROW_SLOT(orig, supd->ripcip);
+            /* Note: supd->ins is never null for column-store. */
+            slot = WT_ROW_SLOT(orig, supd->rip);
```
Comment (keithbostic) [inline src/btree/bt_split.c:1425]: I'd suggest asserting this in the code.

**Review by keithbostic — APPROVED** [review]
lgtm

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/btree/bt_split.c (hunk 1 above)
Comment (sauclovian-wt) [reply src/btree/bt_split.c:1425]: All the places? (I think I commented them all) Or is it sufficient just to put it where where supd is constructed?

Comment (sauclovian-wt) [reply src/btree/bt_split.c:1425]: (conclusion from quick discussion: at the top is sufficient. I've done that)

**Comment by sauclovian-wt** [comment]
Merged the conflict

**Pull request merged**